src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -962,14 +962,20 @@ namespace AppInstaller::CLI::Workflow
         bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
 
         IPackageVersion::Metadata installationMetadata;
+        Utility::Architecture requiredArchitecture = Settings::User().Get<Settings::Setting::InstallArchitectureRequirement>();
         if (isUpdate)
         {
             installationMetadata = context.Get<Execution::Data::InstalledPackageVersion>()->GetMetadata();
         }
         if (context.Args.Contains(Execution::Args::Type::InstallArchitecture))
         {
+            // arguments override settings.
             context.Add<Execution::Data::AllowedArchitectures>({ Utility::ConvertToArchitectureEnum(std::string(context.Args.GetArg(Execution::Args::Type::InstallArchitecture))) });
         }
+        else if (requiredArchitecture != Utility::Architecture::Neutral)

```

Comment:
I would make this an unconditional `else` and the check the settings (both once you revert the `ManifestComparator` changes) inside it.  That way we don't do unnecessary work getting the setting(s) if there is an argument passed in.